geode-core/src/main/java/org/apache/geode/internal/admin/remote/DistributionLocatorId.java

```diff
@@ -317,6 +363,43 @@ public boolean equals(Object other) {
       return false;
     final DistributionLocatorId that = (DistributionLocatorId) other;
 
+    if (this.membername != null && that.membername != null) {
+      if (this.membername.equals(that.membername))
+        return true;
+
+      return false;
+    }
+
+    if (!StringUtils.equals(this.hostnameForClients, that.hostnameForClients))
+      return false;
+    if (this.host != that.host && !(this.host != null && this.host.equals(that.host)))
```

Comment:
I think it should be "||", not "&&".

Reply:
Bruce suggested to use:

if (!Objects.equals(this.host, that.host) {
  return false;
}